components/generalConfig.js

```diff
@@ -0,0 +1,220 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import EditIcon from './icons/edit';
+import CloseIcon from './icons/close';
+import { COLOR, NEW, EDIT, VIEW, DELETE } from '../constants';
+import Modal, { Label, ModalContent, ModalField } from './modal';
+import { firestore, updateHackathonField } from '../utility/firebase';
+
+const KeyValueContainer = styled.div`
+  padding: 5px 0px 5px 5px;
+  cursor: pointer;
+  display: flex;
+  &:hover {
+    background-color: ${COLOR.WHITE};
+  }
+`;
+
+const GeneralKey = styled.span`
+  color: ${COLOR.DARK_GRAY};
+  font-size: 18px;
+  padding-right: 5px;
+`;
+
+const FieldValue = styled.span`
+  font-size: 18px;
+`;
+
+const FieldType = styled.span`
+  color: ${COLOR.DARK_GRAY};
+  font-size: 18px;
+  display: none;
+  ${KeyValueContainer}:hover & {
+    display: block;
+  }
+`;
+
+const HoverContainer = styled.div`
+  margin-left: auto;
+  display: flex;
+`;
+
+const IconContainer = styled.div`
+  padding: 0px 5px 0px 5px;
+`;
+
+const AddField = styled.p`
+  padding: 5px 0px 5px 5px;
+  cursor: pointer;
+  color: ${COLOR.BLUE};
+  font-size: 18px;
+  &:hover {
+    background-color: ${COLOR.WHITE};
+  }
+`;
+
+const TypeSelect = styled.select`
+  color: ${COLOR.BODY_TEXT};
+  width: 100%;
+  text-align: left;
+`;
+
+export default ({ id, title, content }) => {
+  const [field, setField] = useState({});
+  const [isAdding, setIsAdding] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalAction, setModalAction] = useState({});
+
+  const handleAddField = async () => {
+    setField({
+      field: '',
+      type: 'string',
+      value: '',
+    });
+    setIsModalOpen(true);
+    setIsAdding(true);
+    setModalAction(NEW);
+  };
+
+  const handleEditField = async (fieldName) => {
+    setField({
+      field: fieldName,
+      type: typeof content[fieldName],
+      value: content[fieldName],
+    });
+    setIsModalOpen(true);
+    setIsAdding(false);
+    setModalAction(EDIT);
+  };
+
+  const handleClose = async () => {
+    setField({});
+    setIsModalOpen(false);
+  };
+
+  const handleChange = async (property, value) => {
+    setField({
+      ...field,
+      [property]: value,
+    });
+  };
+
+  const handleSave = async (event) => {
+    event.preventDefault();
+    setIsModalOpen(false);
+
+    const fieldKey = `${
+      title === 'globalStyling'
+        ? 'BuildConfig.globalStyling'
+        : `BuildConfig.componentStyling.${title}`
+    }.${field.field}`;
+
+    let fieldValue = field.value;
+
+    if (modalAction === DELETE) {
+      fieldValue = firestore.FieldValue.delete();
+    } else if (field.type === 'number') {
+      // convert to corresponding type if necessary
+      fieldValue = Number(fieldValue);
+      if (!fieldValue) {
+        alert('Field must be a valid number.');
+        return;
+      }
+    } else if (field.type === 'boolean') {
+      fieldValue = fieldValue === 'true';
```

Comment:
```suggestion
      fieldValue = (fieldValue === 'true');
```
Just to make it prettier

Reply:
![image](https://user-images.githubusercontent.com/45786074/108819100-3e2eb480-756f-11eb-8c59-cc4eb78072a7.png)
it goes against the prettier linter LOL